pyglet/event.py

```diff
@@ -421,9 +421,6 @@ def dispatch_event(self, event_type, *args):
             if func:
                 func(*args)
                 return EVENT_HANDLED
-        except AttributeError as exception:
-            self._raise_dispatch_exception(event_type, args, getattr(self, event_type), exception)
-            return EVENT_UNHANDLED
         except TypeError as exception:
```

Comment:
Instead of `TypeError`, we want to catch everything here. `Exception` or `BaseException` is a better choice. If we only catch `TypeError`, other types of exceptions will crash on line 422, and not be passed to `self._raise_dispatch_exception`.

Reply:
Good thinking, as pretty obvious. I'm still not entirely familiar with the whole function chain. So I'm unaware of what not catching the errors will cause other than a hard crash. Modified the code further.